Scripts/UtilityScripts/AzTSUtility.ps1

```diff
@@ -0,0 +1,1617 @@
+# Azure_DataFactory_DP_Avoid_Plaintext_Secrets
+
+function Get-DataFactoryV2()
+{
+  <#
+    .SYNOPSIS
+    This command shows the data factory.
+    .DESCRIPTION
+    This command shows the data factory.
+    .PARAMETER SubscriptionId
+    The Azure subscription ID containing the Data Factory.
+    .PARAMETER ResourceGroupName
+    The Resource Group name containing the Data Factory.
+    .PARAMETER DataFactoryName
+    The Data Factory name.
+    .INPUTS
+    None
+    .OUTPUTS
+    Text with the Data Factory name and its Tag names/values.
+    .EXAMPLE
+    PS> Get-DataFactoryV2 -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -DataFactoryName "MyDataFactoryName"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+      [Parameter(Mandatory=$true)]
+      [string]
+      $SubscriptionId,
+      [Parameter(Mandatory=$true)]
+      [string]
+      $ResourceGroupName,
+      [Parameter(Mandatory=$true)]
+      [string]
+      $DataFactoryName
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  $factory = Get-AzDataFactoryV2 -ResourceGroupName $ResourceGroupName -Name $DataFactoryName
+
+  Write-Information -InformationAction Continue -MessageData ("Data Factory: " + $factory.DataFactoryName)
+  Write-Information -InformationAction Continue -MessageData "Tags:"
+  Write-Information -InformationAction Continue -MessageData $factory.Tags
+}
+
+function Get-DataFactoryV2DataFlows()
+{
+  <#
+    .SYNOPSIS
+    This command lists data flow names and parameters for pipelines that have at least one parameter.
+    .DESCRIPTION
+    This command lists data flow names and parameters for pipelines that have at least one parameter. This is to review parameter values to try and locate parameters that may contain strings identified by the control as plaintext secrets.
+    .PARAMETER SubscriptionId
+    The Azure subscription ID containing the Data Factory.
+    .PARAMETER ResourceGroupName
+    The Resource Group name containing the Data Factory.
+    .PARAMETER DataFactoryName
+    The Data Factory name.
+    .INPUTS
+    None
+    .OUTPUTS
+    For each Data Flow, text with the Data Flow name and its script lines, which begin with parameters and default parameter values.
+    .EXAMPLE
+    PS> Get-DataFactoryV2DataFlows -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -DataFactoryName "MyDataFactoryName"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+      [Parameter(Mandatory=$true)]
+      [string]
+      $SubscriptionId,
+      [Parameter(Mandatory=$true)]
+      [string]
+      $ResourceGroupName,
+      [Parameter(Mandatory=$true)]
+      [string]
+      $DataFactoryName
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  $dataflows = Get-AzDataFactoryV2DataFlow -ResourceGroupName $ResourceGroupName -DataFactoryName $DataFactoryName
```

Comment:
If there are no dataflows in ADF, a simple message like no flows found might be helpful for users.

Reply:
Great point. Added logic to show such a message for all the ADF entity retrieval methods.